### ISSUE TITLE
docker and quagga

### ISSUE THREAD

**Issue by mikenowak** [respawner/looking-glass#18]
Hi @respawner 

I am having a strange issue when connecting to quagga based routers.

Basically I can see the queries being executred on the remote router, but nothing is ever returned and the ajax progress bar keeps on loading forever.

I am running the looking-glass inside a docker container.

I'd appreciate some help how to debug it.

EDIT: I noticed the following error in the Chrome's developer console.

```
Uncaught SyntaxError: Unexpected token < in JSON at position 0
    at Function.parse [as parseJSON] (<anonymous>)
    at Object.<anonymous> (looking-glass.js:88)
    at i (jquery-3.1.1.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-3.1.1.min.js:2)
    at A (jquery-3.1.1.min.js:4)
    at XMLHttpRequest.<anonymous> (jquery-3.1.1.min.js:4)
```

**Comment by respawner** [comment]
So it seems to be an error with the parsing of the result given by the router.
Are you using your down dockerfile? If so, can you share it?

I never really test the looking glass inside a Docker container (not like in real production).
So we might need to make some fixes on the dockerfile provided in the repository.

**Comment by mikenowak** [comment]
I use the "official" image - gmazoyer/looking-glass:latest

The config is as follows

```
<?php

$config['contact']['name'] = 'NOC';
$config['contact']['mail'] = 'email@example.com';

// Frontpage configuration
$config['frontpage']['show_title'] = true;
$config['frontpage']['disclaimer'] = 'All commands will be logged for possible later analysis and statistics.<br />If you don\'t like this policy, please disconnect now!';
$config['frontpage']['show_visitor_ip'] = false;
$config['frontpage']['peering_policy_file'] = 'peering.txt';

// Routers definitions
$config['routers']['core1fra1']['host'] = 'router1.example.com';
$config['routers']['core1fra1']['user'] = 'lg';
$config['routers']['core1fra1']['pass'] = 'password-here';
$config['routers']['core1fra1']['auth'] = 'ssh-password';
$config['routers']['core1fra1']['type'] = 'quagga';
$config['routers']['core1fra1']['source-interface-id']['ipv6'] = '2001:1234:abcd::1';
$config['routers']['core1fra1']['source-interface-id']['ipv4'] = '1.2.3.4';
$config['routers']['core1fra1']['timeout'] = 90;
$config['routers']['core1fra1']['desc'] = 'Router1;

$config['output']['show_command'] = true;
$config['misc']['allow_reserved_ip'] = false;
$config['misc']['allow_private_ip'] = false;
$config['misc']['allow_private_asn'] = false;
```

**Comment by respawner** [comment]
You might be the first one to use it in a production-like environment then :)

I understand your analysis and I think it is the right diagnosis.
So your configuration should not be an issue here.

The only thing for me to try to fix this is:

- reproducing the issue
- trying to patch the dockerfile

If anyone manage to fix this or have a working docker container I'll be happy to accept the contribution :)

**Comment by atypaldos-ias** [comment]
I see this issue on 1.2.1, 1.3.0 and latest.  I also see it running it in docker AND apache on my OSX machine with php56 and apache2.4.29.  I see the client logging into my router, but then disconnecting right away.   I logged in with the read only account just to make sure. Is anyone else experiancing this behavior?  I am running Junos 13.3R8.7

**Comment by atypaldos-ias** [comment]
OK tried the dockerfile and you have to both set the timezone as well as touch the logfile in order for it to write as by default there is no `/var/log/looking-glass.log`  creating a PR with the dockerfile fixes.

**Comment by amtypaldos** [comment]
I see this issue on 1.2.1, 1.3.0 and latest. I also see it running it in docker AND apache on my OSX machine with php56 and apache2.4.29. I see the client logging into my router, but then disconnecting right away. I logged in with the read only account just to make sure. Is anyone else experiancing this behavior? I am running Junos 13.3R8.7

EDIT: I DID NOT SET date.timezone in my php.ini file. Once I set it, fixed the issue. I have not tried with docker yet, I will llet you know.
